File: java/src/main/java/cloud/cniia/openapi/sdk/TransportException.java
package cloud.cniia.openapi.sdk;

/**
 * 传输异常：HTTP/网络错误、超时、非 2xx 状态、响应体非 JSON 等。
 *
 * <p>与 {@link ApiException}（业务码错误）区分：传输层失败抛此异常。
 * 若服务端返回非 2xx，{@link #statusCode()} 为该状态码；纯网络错误（连接失败/超时）时为 -1。
 */
public class TransportException extends RuntimeException {

    private final int statusCode;
    private final String rawBody;

    public TransportException(String message, Throwable cause) {
        super(message, cause);
        this.statusCode = -1;
        this.rawBody = null;
    }

    public TransportException(String message, int statusCode, String rawBody) {
        super(message);
        this.statusCode = statusCode;
        this.rawBody = rawBody;
    }

    /** HTTP 状态码；纯网络错误为 -1。 */
    public int statusCode() {
        return statusCode;
    }

    /** 原始响应体（若有）。 */
    public String rawBody() {
        return rawBody;
    }
}

File: php/src/Exception/TransportException.php
<?php

declare(strict_types=1);

namespace Merchant\Openapi\Exception;

/**
 * 传输异常：网络/cURL 错误、非 2xx HTTP 状态、响应非合法 JSON 信封等。
 *
 * 与 ApiException（业务码非 0）区分：传输层失败时调用方通常需重试或排查网络。
 */
final class TransportException extends \RuntimeException
{
    /**
     * @param string      $message    错误描述
     * @param int|null    $httpStatus HTTP 状态码（如有）
     * @param string|null $rawBody    原始响应体（如有，便于排查）
     */
    public function __construct(
        string $message,
        public readonly ?int $httpStatus = null,
        public readonly ?string $rawBody = null,
        ?\Throwable $previous = null,
    ) {
        parent::__construct($message, 0, $previous);
    }
}
